core/tests/zmq_core_tests/src/zmq_env_test.cpp

```diff
@@ -105,27 +105,28 @@ TEST_F(ZMQEnvTest, UpdateConfigurationTests)
     subscriber2.connect(clientUrl);
     subscriber2.setsockopt(ZMQ_SUBSCRIBE, topic.c_str(), topic.size());
 
-    std::string basename("testfile1.yaml");
+    std::string basename("testfile1.json");
     std::string folderName(TEST_DATA);
     std::string filename = folderName + "/" + basename;
     kpsr::zmq_mdlw::ZMQEnv envTest(filename, "test", topic, 1000, publisher, subscriber, "file1");
 
     std::this_thread::sleep_for(std::chrono::milliseconds(1));
 
-    kpsr::YamlEnvironment yamlEnvPub;
+    kpsr::ConfigurationEnvironment confEnvPub;
 
-    yamlEnvPub.updateConfiguration(
-        "greeting: \"hello\"\ntest: 123\niteration: -1\nkpsr_zmq_env_key: "
-        "\"test\"\nkpsr_zmq_env_topic_name: \"env_data\"\nkpsr_zmq_env_poll_period: 1000",
+    confEnvPub.updateConfiguration(
+        R"({"StringProperties": {"greeting": "hello", "kpsr_zmq_env_key": "test", "kpsr_zmq_env_topic_name": "env_data"},)"
+        R"( "IntProperties": {"test": 123, "iteration": -1, "kpsr_zmq_env_poll_period": 1000}})",
```

Comment:
Classified by data type,
```json
{
  "StringProperties": {
    "greeting": "hello",
    "kpsr_zmq_env_key": "test",
    "kpsr_zmq_env_topic_name": "env_data"
  },
  "IntProperties": {
    "test": 123,
    "iteration": -1,
    "kpsr_zmq_env_poll_period": 1000
  }
}
```
as in `core/tests/data/*.json`. It could/should be
```json
{
  "greeting": "hello",
  "kpsr_zmq_env_key": "test",
  "kpsr_zmq_env_topic_name": "env_data",
  "test": 123,
  "iteration": -1,
  "kpsr_zmq_env_poll_period": 1000
}
```

Reply:
For the environment, we cannot do that, because it has to be generic.